### ISSUE TITLE
Deprecated API usage

### ISSUE THREAD

**Issue by ThexXTURBOXx** [mogol/flutter_secure_storage#435]
It seems as if some APIs have been deprecated again. I am using Android SDK `33`

```
C:\Users\Nico Mexis\AppData\Local\Pub\Cache\hosted\pub.dartlang.org\flutter_secure_storage-6.0.0\android\src\main\java\com\it_nomads\fluttersecurestorage\ciphers\RSACipher18Implementation.java:133: warning: [deprecation] updateConfiguration(Configuration,DisplayMetrics) in Resources has been deprecated
        context.getResources().updateConfiguration(config, context.getResources().getDisplayMetrics());
                              ^
C:\Users\Nico Mexis\AppData\Local\Pub\Cache\hosted\pub.dartlang.org\flutter_secure_storage-6.0.0\android\src\main\java\com\it_nomads\fluttersecurestorage\ciphers\RSACipher18Implementation.java:137: warning: [deprecation] locale in Configuration has been deprecated
        config.locale = locale;
              ^
C:\Users\Nico Mexis\AppData\Local\Pub\Cache\hosted\pub.dartlang.org\flutter_secure_storage-6.0.0\android\src\main\java\com\it_nomads\fluttersecurestorage\ciphers\RSACipher18Implementation.java:171: warning: [deprecation] KeyPairGeneratorSpec in android.security has been deprecated
        return new android.security.KeyPairGeneratorSpec.Builder(context)
                                   ^
C:\Users\Nico Mexis\AppData\Local\Pub\Cache\hosted\pub.dartlang.org\flutter_secure_storage-6.0.0\android\src\main\java\com\it_nomads\fluttersecurestorage\ciphers\RSACipher18Implementation.java:171: warning: [deprecation] Builder in KeyPairGeneratorSpec has been deprecated
        return new android.security.KeyPairGeneratorSpec.Builder(context)
                                                        ^
C:\Users\Nico Mexis\AppData\Local\Pub\Cache\hosted\pub.dartlang.org\flutter_secure_storage-6.0.0\android\src\main\java\com\it_nomads\fluttersecurestorage\ciphers\RSACipher18Implementation.java:171: warning: [deprecation] Builder(Context) in Builder has been deprecated
        return new android.security.KeyPairGeneratorSpec.Builder(context)
               ^
C:\Users\Nico Mexis\AppData\Local\Pub\Cache\hosted\pub.dartlang.org\flutter_secure_storage-6.0.0\android\src\main\java\com\it_nomads\fluttersecurestorage\ciphers\RSACipher18Implementation.java:172: warning: [deprecation] setAlias(String) in Builder has been deprecated
                .setAlias(keyAlias)
                ^
C:\Users\Nico Mexis\AppData\Local\Pub\Cache\hosted\pub.dartlang.org\flutter_secure_storage-6.0.0\android\src\main\java\com\it_nomads\fluttersecurestorage\ciphers\RSACipher18Implementation.java:173: warning: [deprecation] setSubject(X500Principal) in Builder has been deprecated
                .setSubject(new X500Principal("CN=" + keyAlias))
                ^
C:\Users\Nico Mexis\AppData\Local\Pub\Cache\hosted\pub.dartlang.org\flutter_secure_storage-6.0.0\android\src\main\java\com\it_nomads\fluttersecurestorage\ciphers\RSACipher18Implementation.java:174: warning: [deprecation] setSerialNumber(BigInteger) in Builder has been deprecated
                .setSerialNumber(BigInteger.valueOf(1))
                ^
C:\Users\Nico Mexis\AppData\Local\Pub\Cache\hosted\pub.dartlang.org\flutter_secure_storage-6.0.0\android\src\main\java\com\it_nomads\fluttersecurestorage\ciphers\RSACipher18Implementation.java:175: warning: [deprecation] setStartDate(Date) in Builder has been deprecated
                .setStartDate(start.getTime())
                ^
C:\Users\Nico Mexis\AppData\Local\Pub\Cache\hosted\pub.dartlang.org\flutter_secure_storage-6.0.0\android\src\main\java\com\it_nomads\fluttersecurestorage\ciphers\RSACipher18Implementation.java:176: warning: [deprecation] setEndDate(Date) in Builder has been deprecated
                .setEndDate(end.getTime())
                ^
C:\Users\Nico Mexis\AppData\Local\Pub\Cache\hosted\pub.dartlang.org\flutter_secure_storage-6.0.0\android\src\main\java\com\it_nomads\fluttersecurestorage\ciphers\RSACipher18Implementation.java:177: warning: [deprecation] build() in Builder has been deprecated
                .build();
                ^
11 warnings
```

**Comment by JoseBarreto1** [comment]
Note: /home/ti/snap/flutter/common/flutter/.pub-cache/hosted/pub.dartlang.org/flutter_secure_storage-6.0.0/android/src/main/java/com/it_nomads/fluttersecurestorage/ciphers/RSACipher18Implementation.java uses or overrides a deprecated API.
Note: Recompile with -Xlint:deprecation for details.

**Comment by VasuBhav** [comment]
Facing Same Problem with flutter 3.3.4

**Comment by juliansteenbakker** [comment]
This is fixed with version 6.1.0
